sled-agent/src/hardware/illumos/sysconf.rs

```diff
@@ -0,0 +1,40 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Access to sysconf-related info.
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("sysconf failed accessing {arg}: {e}")]
+    Sysconf { arg: String, e: std::io::Error },
+
+    #[error("Integer conversion error: {0}")]
+    Integer(#[from] std::num::TryFromIntError),
+}
+
+fn sysconf(argname: &str, arg: libc::c_int) -> Result<u64, Error> {
```

Comment:
Just want to make sure its intentional that we're doing i64->u64 conversion on the return value here. I assume that we're relying on negative values only being returned for the `-1 case. But if that's not the case, won't we end up converting something to a large unsigned value?

If we're relying on this we should probably be a little defensive as other consumers are going to heartbeat this into the database if we have something wrong.

Reply:
I guess the try info may make this safe and I misunderstood that due to make lack of familiarity. I might be explicit about why we're doing the type conversions somewhere to explain to a reader why we want the uniform u64.